application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -400,6 +433,19 @@ private TransportApiResponseMsg getTransportApiResponseMsg(DeviceCredentials dev
                 .build();
     }
 
+    private ListenableFuture<TransportApiResponseMsg> handle(GetSnmpDevicesRequestMsg requestMsg) {
+        List<UUID> result = deviceService.findDevicesIdsByDeviceProfileTransportType(DeviceTransportType.SNMP);
```

Comment:
please add "TODO" to implement this via pagination. This may not work with hundreds of thousands of devices.